transport/src/main/java/io/netty/channel/pool/FixedChannelPool.java

```diff
@@ -290,32 +295,36 @@ private void tooManyOutstanding(Promise<?> promise) {
 
             @Override
             public void operationComplete(Future<Void> future) throws Exception {
-                assert executor.inEventLoop();
-
-                if (closed) {
-                    // Since the pool is closed, we have no choice but to close the channel
-                    channel.close();
-                    promise.setFailure(new IllegalStateException("FixedChannelPool was closed"));
-                    return;
-                }
+                try {
+                    assert executor.inEventLoop();
+
+                    if (closed) {
+                        // Since the pool is closed, we have no choice but to close the channel
+                        channel.close();
+                        promise.setFailure(new IllegalStateException("FixedChannelPool was closed"));
+                        return;
+                    }
 
-                if (future.isSuccess()) {
-                    decrementAndRunTaskQueue();
-                    promise.setSuccess(null);
-                } else {
-                    Throwable cause = future.cause();
-                    // Check if the exception was not because of we passed the Channel to the wrong pool.
-                    if (!(cause instanceof IllegalArgumentException)) {
+                    if (future.isSuccess()) {
                         decrementAndRunTaskQueue();
+                        promise.setSuccess(null);
+                    } else {
+                        Throwable cause = future.cause();
+                        // Check if the exception was not because of we passed the Channel to the wrong pool.
+                        if (!(cause instanceof IllegalArgumentException)) {
+                            decrementAndRunTaskQueue();
+                        }
+                        promise.setFailure(future.cause());
                     }
-                    promise.setFailure(future.cause());
+                } catch (Throwable cause) {
+                    promise.setFailure(cause);
```

Comment:
This should use tryFailure() as we may be here if setFailure() threw above

Reply:
Done